python/openassetio_traitgen/generators/helpers.py

```diff
@@ -73,9 +73,23 @@ def package_dependencies(
     """
     dependencies = set()
     for declaration in declarations:
-        if isinstance(declaration, SpecificationDeclaration):
-            for trait in declaration.trait_set:
-                dependencies.add(trait.package)
+        dependencies |= set(declaration_dependencies(declaration))
+    dependencies = list(dependencies)
+    dependencies.sort()
+    return dependencies
+
+
+def declaration_dependencies(
+    declaration: Union[SpecificationDeclaration, TraitDeclaration]
+) -> List[str]:
+    """
```

Comment:
It is a bit of a misnomer. The "package" is actually a namespace (within a package, with the `declarations` list being the members of the namespace).

Perhaps it should then be `namespace_dependencies` and `specification_dependencies`? :thinking:

Reply:
Oh perhaps `package_dependencies` was intended to flag the return type, rather than input type, i.e. the packages the input depends on.  That makes some sense.  Which does make `declaration_dependencies` the misnomer.

Reply:
Renamed as part of refactor in d62b361f